### DISCUSSION TITLE
Binance : error getting response -JSON

### DISCUSSION THREAD

**Discussion by takai-sama** [grafana/grafana-infinity-datasource#224] [Specific APIs] [answered]
hello, thanks for the plugin. it's nice job. i want to get API from api.binance.com but when i add the url in the plugin, it's getting error response like this. (for example i want to get api from [https://api.binance.com/api/v3/klines?symbol=BNBBTC&interval=1m](url) )
![image](https://user-images.githubusercontent.com/77669943/147456897-a590e262-82ad-48a0-a941-a1ea541ed84f.png)
it must be like this
![image](https://user-images.githubusercontent.com/77669943/147457516-37f0ea67-e1a7-4a76-9b54-8d9b1fb11a9e.png)
i tried another link from binance but still get error.
any idea?

**Comment by yesoreyeram** [comment, marked as answer]
Not sure what's going on with your instance. But I am able to retrieve and visualize the data with latest version. I used the API with format https://api.binance.com/api/v3/klines?symbol=BNBBTC&interval=1m&startTime=$__from&endTime=$__to

![image](https://user-images.githubusercontent.com/153843/147469718-1a07c707-ade8-4280-bb46-da7b91d7928c.png)

**Reply by omalisson** [reply]
Top!

**Reply by takai-sama** [reply]
this is very embarrassing, i just realized that the binance website is blocked in my country. i can open it in chrome because i use openDNS, but grafana server can't access it. thanks for fast response, sorry for my mistake

**Comment by takai-sama** [comment]
my problem with error getting response is solved using dnscrypt.
pardon me @yesoreyeram , but how to add variable on url ? this way doesn't work 
![image](https://user-images.githubusercontent.com/77669943/147630012-81a84067-ea67-4399-b5e1-c180d53e13e0.png)
i tried in query parameter, adding variabel as value on symbol as key still doesn't work

**Reply by yesoreyeram** [reply]
It should work that way. I guess you might be referring variable display name instead name. 

In case if that doest help, try updating the plugin to latest beta.0.8. ,😳

**Reply by yesoreyeram** [reply]
Also remember you must add the query string either in url or in query params.only one place. Both are same way..
